ports/esp32/README.md

```diff
@@ -50,10 +50,16 @@ To check out a copy of the IDF use git clone:
 $ git clone -b v4.0.2 --recursive https://github.com/espressif/esp-idf.git
 ```
 
-You can replace `v4.0.2` with `v4.1.1` or any other supported version.
+You can replace `v4.0.2` with `v4.1.1` or `v4.2` or any other supported version.
 (You don't need a full recursive clone; see the `ci_esp32_setup` function in
 `tools/ci.sh` in this repository for more detailed set-up commands.)
 
+Use these commands only if the ESP-IDF has been cloned before without the '--recursive' option.
+```bash
+$ cd esp-idf
+$ git submodule update --init --recursive
```

Comment:
If this command is needed then something went wrong with the initial clone of the repository, and it's not really feasible to list in this README all the ways to fix the cloning process.  If you followed the instructions exactly as in this README then there should not be any issue.

But I understand that a lot of people will already have the IDF checked out for some other reason, so it'd make sense to provide some simple instructions here for such a case.  Eg:
```
If you already have a copy of the IDF then checkout a version compatible with
MicroPython and update the submodules using:

$ cd esp-idf
$ git checkout v4.2
$ git submodule update --init --recursive
```